src/gausskernel/process/tcop/postgres.cpp

```diff
@@ -7316,6 +7320,9 @@ int PostgresMain(int argc, char* argv[], const char* dbname, const char* usernam
     else
         t_thrd.proc_cxt.PostInit->InitBackendWorker();
 
+    /* Connect to K2PG cluster. */
+	K2PgInitPostgresBackend("postgres", dbname, username);
```

Comment:
This is only called once for the whole lifetime right? Whereas before it was called after every fork for each new connection. As part of this call we call PgGate_InitSession, so that probably needs to move somewhere else? I'm not sure what work InitSession needs to do or even if we need it at all.

Reply:
PG gate will be initialized only once. But PgGate_InitSession() will be called for each new session for each thread in the connection pool. I will add logic to call PgGate_InitSession() for thread connections, sorry for not clarifying that.

The PgGate_InitSession() in the K2PgInitPostgresBackend call here is a default connection session with the default database.

Reply:
I see, thanks!